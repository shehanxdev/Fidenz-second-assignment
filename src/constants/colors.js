export const COLORS = [
  "#388EE7",
  "#8B49CC",
  "#3C882C",
  "#C63F3F",
  "#6249CC",
  "#1B127B",
  "#D847DB",
  "#138E78",
  "#FF5733",
  "#66CC99",
  "#E74C3C",
  "#F39C12",
  "#3498DB",
  "#9B59B6",
  "#27AE60",
  "#F1C40F",
];
