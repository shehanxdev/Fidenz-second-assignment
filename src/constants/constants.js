export const CARD_COLORS = {
  grey: "#585a61",
  skyBlue: "#388EE7",
  green: "#40B681",
  purple: "#6249CC",
  orange: "#DE944E",
  red: "#9C3A3A",
  black: "#1F2128",
  white: "#e7e7e7",
};
//*To be access using index
export const CARD_COLORS_ARRAY = [
  CARD_COLORS.skyBlue,
  CARD_COLORS.green,
  CARD_COLORS.orange,
  CARD_COLORS.purple,
  CARD_COLORS.red,
];

export const LOCAL_STORAGE_KEY = "cachedWeatherData";

export const expirationTime = 5 * 60 * 1000;
