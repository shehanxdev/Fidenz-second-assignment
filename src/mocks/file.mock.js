export const mockWeatherData = {
  coord: { lon: -74.006, lat: 40.7128 },
  sunrise: 1642396800,
  sunset: 1642436799,
  country: "US",
  description: "Clear sky",
  icon: "01d",
  temp: 25.5,
  pressure: 1015,
  humidity: 50,
  temp_min: 22.0,
  temp_max: 28.0,
  visibility: 10000,
  speed: 5.0,
  deg: 180,
  all: 1,
  dt: 1642422236,
  id: 5128581,
  name: "New York",
};
