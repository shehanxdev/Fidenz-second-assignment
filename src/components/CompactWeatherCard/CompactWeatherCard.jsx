import React, { useState } from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faClose } from "@fortawesome/free-solid-svg-icons";
import { formatDate } from "../../utilities";
import { CARD_COLORS } from "../../constants";
import { CARDBACKGROUND } from "../../images";
import { WeatherCardFooter } from "../WeatherCardFooter";

export function CompactWeatherCard({ weatherData, bgColor }) {
  const customCardStyles = {
    backgroundColor: bgColor,
    backgroundImage: `url(${CARDBACKGROUND})`,
    backgroundSize: "cover",
    backgroundRepeat: "no-repeat",
    backgroundPositionX: "center",
  };
  const navigateToExpandedCardView = () => {
    alert("hello");
  };

  return (
    <div
      className=" text-light d-flex flex-column rounded-3 overflow-hidden"
      onClick={navigateToExpandedCardView}
    >
      <div className="p-2" style={customCardStyles}>
        <div className="d-flex justify-content-end">
          <FontAwesomeIcon icon={faClose} style={{ color: "white" }} />
        </div>
        <div className="m-1 m-md-4 px-1 d-flex justify-content-between">
          <div className="d-flex flex-column justify-content-between">
            <div className="d-flex flex-column">
              <span className="fs-3">
                {" "}
                {`${weatherData.name}, ${weatherData.country}`}{" "}
              </span>
              <span className="align-self-center">{formatDate()}</span>
            </div>
            <div className="align-self-center fs-5">
              {weatherData.description}
            </div>
          </div>
          <div className="d-flex flex-column justify-content-between ">
            <span className="d-flex flex-column display-1 ">
              <strong>{weatherData.temp.toFixed(0)}&deg;c</strong>
            </span>
            <div>
              <span className="d-block">
                Temp Min: {weatherData.temp_min}&deg;c
              </span>
              <span className="d-block">
                Temp Max: {weatherData.temp_max}&deg;c
              </span>
            </div>
          </div>
        </div>
      </div>

      <WeatherCardFooter weatherData={weatherData} />
    </div>
  );
}
