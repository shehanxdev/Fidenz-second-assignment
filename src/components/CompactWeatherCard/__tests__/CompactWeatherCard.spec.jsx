import React from "react";
import renderer from "react-test-renderer";
import { mockWeatherData } from "../../../mocks";
import { CARD_COLORS_ARRAY } from "../../../constants";
import { MemoryRouter } from "react-router-dom";
import { CompactWeatherCard } from "./../CompactWeatherCard";

describe("WeatherCardFooter component", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <CompactWeatherCard
            weatherData={mockWeatherData}
            bgColor={CARD_COLORS_ARRAY[0]}
          />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
