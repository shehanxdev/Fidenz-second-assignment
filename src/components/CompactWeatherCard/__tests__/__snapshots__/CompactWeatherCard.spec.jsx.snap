// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCardFooter component renders correctly 1`] = `
<div
  className=" text-light d-flex flex-column rounded-3 overflow-hidden"
  onClick={[Function]}
>
  <div
    className="p-3"
    style={
      Object {
        "backgroundColor": "#388EE7",
        "backgroundImage": "url(image-file-stub)",
        "backgroundPositionX": "center",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "cover",
      }
    }
  >
    <div
      className="d-flex justify-content-end"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-xmark "
        data-icon="xmark"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={
          Object {
            "color": "white",
          }
        }
        viewBox="0 0 384 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="m-1 m-md-4 px-1 d-flex justify-content-between"
    >
      <div
        className="d-flex flex-column justify-content-between"
      >
        <div
          className="d-flex flex-column align-items-center"
        >
          <span
            className="fs-3"
          >
             
            New York, US
             
          </span>
          <span
            className="align-self-center"
          >
            04:51 pm, Jan 28
          </span>
        </div>
        <div
          className="align-self-center fs-5"
        >
          <img
            alt="weather-icon"
            src="http://openweathermap.org/img/w/01d.png"
          />
          Clear sky
        </div>
      </div>
      <div
        className="d-flex flex-column justify-content-between "
      >
        <span
          className="d-flex flex-column display-1 "
        >
          <strong>
            26
            °c
          </strong>
        </span>
        <div>
          <span
            className="d-block"
          >
            Temp Min: 
            22
            °c
          </span>
          <span
            className="d-block"
          >
            Temp Max: 
            28
            °c
          </span>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="d-flex justify-content-center gap-3 p-4 "
      style={
        Object {
          "backgroundColor": "#383B47",
        }
      }
    >
      <div
        className="d-flex flex-column"
      >
        <span>
          Pressure: 1015hPa
        </span>
        <span>
          Humidity: 50%
        </span>
        <span>
          Visibility: 10km
        </span>
      </div>
      <div
        style={
          Object {
            "border": "1px solid #585a61",
          }
        }
      />
      <div
        className="d-flex flex-column justify-content-center gap-2"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-location-arrow fa-2x "
          data-icon="location-arrow"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={
            Object {
              "color": "white",
            }
          }
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M429.6 92.1c4.9-11.9 2.1-25.6-7-34.7s-22.8-11.9-34.7-7l-352 144c-14.2 5.8-22.2 20.8-19.3 35.8s16.1 25.8 31.4 25.8H224V432c0 15.3 10.8 28.4 25.8 31.4s30-5.1 35.8-19.3l144-352z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <span>
          5.0m/s 180 Degree
        </span>
      </div>
      <div
        style={
          Object {
            "border": "1px solid #585a61",
          }
        }
      />
      <div
        className="d-flex flex-column justify-content-center"
      >
        <span>
          Sunrise: 6:05am
        </span>
        <span>
          Sunset: 6:05am
        </span>
      </div>
    </div>
  </div>
</div>
`;
